erpnext/accounts/doctype/sales_invoice/sales_invoice.js

```diff
@@ -568,7 +568,7 @@ cur_frm.fields_dict.write_off_cost_center.get_query = function(doc) {
 cur_frm.set_query("income_account", "items", function(doc) {
 	return{
 		query: "erpnext.controllers.queries.get_income_account",
-		filters: {'company': doc.company}
+		filters: {'company': doc.company, "disabled": 0}
```

Comment:
While you are doing this, maybe remove usage or cur_frm and do the standard way using from triggers?